client/common/cmdline.c

```diff
@@ -48,6 +48,102 @@
 #include <freerdp/log.h>
 #define TAG CLIENT_TAG("common.cmdline")
 
+static BOOL freerdp_path_valid(const char* path, BOOL* special)
+{
+	const char DynamicDrives[] = "DynamicDrives";
+	BOOL isPath = FALSE;
+	BOOL isSpecial;
+	if (!path)
+		return FALSE;
+
+	isSpecial = (strncmp(path, "*", 2) == 0) ||
+						(strncmp(path, DynamicDrives, sizeof(DynamicDrives)) == 0) ||
+						(strncmp(path, "%", 2) == 0) ? TRUE : FALSE;
+	if (!isSpecial)
+		isPath = PathFileExistsA(path);
+
+	if (special)
+		*special = isSpecial;
+
+	return isSpecial || isPath;
+}
+
+static BOOL freerdp_sanitize_drive_name(char* name, const char* invalid, const char* replacement)
+{
+	if (!name || !invalid || !replacement)
+		return FALSE;
+	if (strlen(invalid) != strlen(replacement))
+		return FALSE;
+
+	while(*invalid != '\0')
+	{
+		const char what = *invalid++;
+		const char with = *replacement++;
+
+		char* cur = name;
+		while((cur = strchr(cur, what)) != NULL)
+			*cur = with;
+	}
+	return TRUE;
+}
+
+static BOOL freerdp_client_add_drive(rdpSettings* settings, const char* path, const char* name)
+{
+	RDPDR_DRIVE* drive;
+
+	drive = (RDPDR_DRIVE*) calloc(1, sizeof(RDPDR_DRIVE));
+
+	if (!drive)
+		return FALSE;
+
+	drive->Type = RDPDR_DTYP_FILESYSTEM;
+
+	if (name)
+	{
+		/* Path was entered as secondary argument, swap */
+		if (PathFileExistsA(name))
+		{
+			const char* tmp = path;
+			path = name;
+			name = tmp;
+		}
+	}
+
+	if (name)
+	{
+		if (!(drive->Name = _strdup(name)))
+			goto fail;
+	}
+	else /* We need a name to send to the server. */
+		if (!(drive->Name = _strdup(path)))
+			goto fail;
+
+	if (!path || !freerdp_sanitize_drive_name(drive->Name, "\\/:", "___"))
```

Comment:
The : should not be sanitized/replaced with _

Reply:
are you sure? a name with a `:` might otherwise be interpreted as path.